ezufo_helpers/find_360_overlap.py

```diff
@@ -28,116 +29,213 @@ def parse_args():
+        if '.tif' in filename:
+            filenames.append(filename)
     filenames.sort()
 
+    # read a sample image to determine image dimensions
+    sample_img = read_image(os.path.join(dir_name, filenames[0])).astype(np.uint16)
+    # allocate empty np array to hold the image stack, then fill the array
+    sequence = np.zeros([len(filenames), 2*del_crop, sample_img.shape[1]])
+    i = 0
     for filename in filenames:
-        if '.tif' in filename:
-            tif_img = read_image(os.path.join(dir_name, filename)).astype(np.uint16)
-            sequence.append(np.array(tif_img)[(row-10):(row+10),:])
+        tif_img = read_image(os.path.join(dir_name, filename)).astype(np.uint16)
+        sequence[i] = tif_img[(row-del_crop):(row+del_crop),:]
+        i = i+1
     return np.array(sequence)
 
 
 def main():
 
     # assign parsed command line arguments to variables
     args = parse_args()
-
     root = args.input
     proc = args.tmpdir
     output =  args.output
     row_num = args.row
     overlap_min = args.min
     overlap_max = args.max
     overlap_increment = args.step
-    axis_on_left = args.axis_on_left
-
-    # recursively create output temporary directory if it doesn't exist
-    if os.path.exists(os.path.join(proc)):
-        shutil.rmtree(proc)
-        os.makedirs(os.path.join(proc, 'sinos'))
-    else:
-        os.makedirs(os.path.join(proc, 'sinos'))
-
-
-
-    # concatenate images end-to-end and generate a sinogram
-    print('opening half-acquisition image sequence...')
-    tomo = open_tif_sequence(os.path.join(root,'tomo'), row_num)
-
-    #open flats and darks and average them
-    flat = np.mean(open_tif_sequence(os.path.join(root,'flats'), row_num)/65535.0, axis=0)
-    flat2 = np.mean(open_tif_sequence(os.path.join(root,'flats2'), row_num)/65535.0, axis=0)
-    dark = np.mean(open_tif_sequence(os.path.join(root,'darks'), row_num)/65535.0, axis=0)
-
-    tomo_single_row = tomo[:,tomo.shape[1]//2,:]/65535.0
-    dark_single_row = np.tile(dark[tomo.shape[1]//2,:], (tomo.shape[0],1))
-    flat_single_row = np.zeros((tomo.shape[0], tomo.shape[2]))
-    img_height = tomo.shape[0]
-    img_width = tomo.shape[1]
-
-    del tomo
-
-    #create interpolated sinogram of flats on the same row as we use for the projections, then carry out flat/dark correction
-    print('creating stitched sinograms...')
-    for i in range(0, img_height):
-        flat_single_row[i,:] = (flat[img_width//2,:]*(float(i)/float(img_height))+flat2[img_width//2,:]*(1.0 - float(i)/float(img_height)))
-
-    tomo_sino_corr = (tomo_single_row - dark_single_row) / (flat_single_row - dark_single_row)
-    max_gray_value = tomo_sino_corr.max()
-    tomo_sino_corr = tomo_sino_corr/max_gray_value
-
-    tomo_first_half = tomo_sino_corr[:int(tomo_sino_corr.shape[0]/2),:]
-    tomo_second_half = tomo_sino_corr[int(tomo_sino_corr.shape[0]/2):tomo_sino_corr.shape[0],:]
-
-    tomo_first_half = -1.0*np.log(tomo_first_half)
```

Comment:
Toby, to make it resilient to possible non-positive number we need to remove Nan and Infinities
expression = '(isinf (v) || isnan (v) || (v <= 0)) ? 0.0f : -log ({} * v) * {{}}'